test/cpp/api/imethod.cpp

```diff
@@ -8,30 +8,40 @@
 using namespace ::testing;
 using namespace caffe2;
 
-// TODO(T96218435): Enable the following tests in OSS.
+const char* simple = "torch/csrc/deploy/example/generated/simple";
+const char* simpleJit = "torch/csrc/deploy/example/generated/simple_jit";
+
+// TODO(jwtan): Try unifying cmake and buck for getting the path.
+const char* path(const char* envname, const char* path) {
+  const char* e = getenv(envname);
+  return e ? e : path;
+}
+
+// TODO(jwtan): Considering moving the following tests to somewhere else as linking deploy (sub-classes)
```

Comment:
i think it's ok as is for now.

but if you wanted to change it, I think maybe what I'd do is,
- always add imethod.cpp into TORCH_API_TEST_SOURCES
- use a preprocessor directive like `USE_DEPLOY` (idk if we even have one, but we could add one) to conditionally add the parts to the unit tests that use deploy. 

That way the scriptmethod part would still be tested always, and, i think logically the tests _do_ belong here with the other API tests

Reply:
For the script method, this is definitely the right place. For the deploy, I guess I'm confused by the current state where it is a standalone library other than torch. Would it in the future be merged into the torch library? Or is it logically a part of the torch library but it requires special treatment to do the custom loader magic? (I'm okay to remove the comment.)